core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -187,19 +193,24 @@ public byte asByte() {
     }
 
     public double asDouble() {
-        if (this == AUTO) {
+        if (isAuto()) {
             return 1d;
         }
         return Double.parseDouble(fuzziness.toString());
     }
 
     public float asFloat() {
-        if (this == AUTO) {
+        if (isAuto()) {
             return 1f;
         }
         return Float.parseFloat(fuzziness.toString());
     }
 
+    // A deserialized fuzziness with AUTO is not the static Fuzziness.AUTO object
+    private boolean isAuto() {
+        return this == AUTO || fuzziness.equalsIgnoreCase("AUTO");
```

Comment:
Here the equals means that the current Object is actually the AUTO Object which happens when parsing in Fuzziness#parse. So either we have the AUTO Object or we have a new Object that has a fuzziness with "AUTO".